### PR TITLE
Add app.Err() to expose the underlying error from New

### PR DESCRIPTION
**Pull request by glibsm**
PR against #551 to try out what `app.Err()` would look and feel like.

### CODE REVIEW

File: app.go [hunk 1]
```diff
@@ -211,6 +211,15 @@ func New(opts ...Option) *App {
 	return app
 }
 
+// Error returns an error that may have been encountered during the
+// graph resolution.
+//
+// This includes things like incomplete graphs, circular dependencies,
+// missing dependencies, invalid constructors, etc.
+func (app *App) Error() error {
```
Comment (alsamylkin) [inline app.go]: I'd really prefer Err(), because Error() usually returns a string, implementing the error interface..

File: app_test.go [hunk 2]
```diff
@@ -240,8 +252,13 @@ func TestAppStart(t *testing.T) {
 			Invoke(Invoke(func(type1) {
 			})),
 		)
+		newErr := app.Err()
+		require.Error(t, newErr)
+
 		err := app.Start(context.Background())
 		require.Error(t, err, "expected start failure")
+		assert.Equal(t, err, newErr, "start should return the same error fx.New encountered")
```
Comment (alsamylkin) [inline app_test.go]: should run return the same error too?

Comment (abhinav) [inline app_test.go]: Run just panics/exits, right? I think that's preferable. Keeps the default `main()` the same.

Comment (alsamylkin) [inline app_test.go]: Yeah, makes sense.

**Pull request merged**